internal/kafka/utils.go

```diff
@@ -134,6 +134,13 @@ func getCmkEncryptionKey(cluster *cmkv2.CmkV2Cluster) string {
 	return ""
 }
 
+func getCmkZones(cluster *cmkv2.CmkV2Cluster) []string {
+	if isDedicated(cluster) && cluster.GetSpec().Config.CmkV2Dedicated.GetZones() != nil {
+		return cluster.GetSpec().Config.CmkV2Dedicated.GetZones()
+	}
+	return []string{}
+}
+
```

Comment:
Again, `isDedicated` already checks the nil pointer, and `cluster.GetSpec().Config.CmkV2Dedicated.GetZones()` will never be nil.

```suggestion
func getCmkZones(cluster *cmkv2.CmkV2Cluster) []string {
	if isDedicated(cluster) && cluster.Spec.Config.CmkV2Dedicated.Zones != nil {
		return cluster.Spec.Config.CmkV2Dedicated.GetZones()
	}
	return []string{}
}

```